.github/workflows/track_benches.yml

```diff
@@ -0,0 +1,95 @@
+      PR_BENCHMARK_RESULTS: pr_${{ matrix.backend }}.txt
+      BASE_BENCHMARK_RESULTS: base_${{ matrix.backend }}.txt
+      GITHUB_EVENT: event_${{ matrix.backend }}.json
+    steps:
+      - name: Download Benchmark Results
+        uses: actions/github-script@v6
+        with:
+          script: |
+            async function downloadArtifact(artifactName) {
+              let allArtifacts = await github.rest.actions.listWorkflowRunArtifacts({
+                owner: context.repo.owner,
+                repo: context.repo.repo,
+                run_id: context.payload.workflow_run.id,
+              });
+              let matchArtifact = allArtifacts.data.artifacts.filter((artifact) => {
+                return artifact.name == artifactName
+              })[0];
+              if (!matchArtifact) {
+                core.setFailed(`Failed to find artifact: ${artifactName}`);
+              }
+              let download = await github.rest.actions.downloadArtifact({
+                owner: context.repo.owner,
+                repo: context.repo.repo,
+                artifact_id: matchArtifact.id,
+                archive_format: 'zip',
+              });
+              let fs = require('fs');
+              fs.writeFileSync(`${process.env.GITHUB_WORKSPACE}/${artifactName}.zip`, Buffer.from(download.data));
+            }
+            await downloadArtifact(process.env.PR_BENCHMARK_RESULTS);
+            await downloadArtifact(process.env.BASE_BENCHMARK_RESULTS);
+            await downloadArtifact(process.env.GITHUB_EVENT);
+      - name: Unzip Benchmark Results
+        run: |
+          unzip $PR_BENCHMARK_RESULTS.zip
+          unzip $BASE_BENCHMARK_RESULTS.zip
+          unzip $GITHUB_EVENT.zip
+      - name: Export GitHub Event Data
+        uses: actions/github-script@v6
+        with:
+          script: |
+            let fs = require('fs');
+            let githubEvent = JSON.parse(fs.readFileSync("event.json", {encoding: 'utf8'}));
+            console.log(githubEvent);
+            core.exportVariable("PR_HEAD", `${githubEvent.pull_request.head.ref}-${githubEvent.pull_request.head.sha.slice(0, 8)}`);
+            core.exportVariable("PR_ID", `${githubEvent.pull_request.head.ref}/${process.env.BENCHER_TESTBED}/${process.env.BENCHER_ADAPTER}`);
+            core.exportVariable("PR_NUMBER", githubEvent.number);
+      - uses: bencherdev/bencher@main
+      - name: Track base Benchmarks
+        run: |
+          bencher run \
+          --if-branch '${{ env.PR_HEAD }}' \
+          --else-branch \
+          --token "${{ secrets.BENCHER_API_TOKEN }}" \
+          --file "$BASE_BENCHMARK_RESULTS"
+      - name: Create PR threshold
+        run: |
+          bencher threshold create \
+          --project "$BENCHER_PROJECT" \
+          --branch '${{ env.PR_HEAD }}' \
+          --testbed "$BENCHER_TESTBED" \
+          --measure latency \
+          --test t \
+          --upper-boundary 0.98 \
```

Comment:
That's essentially the p value for the test to reject the PR if the performance changes are to large. (So if it's outside 0.98 of the assumed distribution, right). Might be worth to add a short comment there just in case we want to touch that value later on.

Reply:
Yes, and great point!
I pulled this out to an env var and added a comment to make it clearer 😃